utilities/transactions/transaction_base.h

```diff
@@ -330,8 +330,14 @@ class TransactionBaseImpl : public Transaction {
   // Optimistic Transactions will keep note the requested locks (not actually
   // locked), and do conflict checking until commit time based on the tracked
   // lock requests.
+  //
+  // Declared as public because
+  //  - RangeLocking's lock escalation may replace it
```

Comment:
I didn't see that it's replaced anywhere?

Reply:
Ok, the pointed object (RangeTreeLockTracker) is not replaced, but it's contents are modified in `RangeTreeLockManager::on_escalate`  (under a mutex):
https://github.com/spetrunia/rocksdb/blob/1dfeb2bfa089adc78378459b40650b5b5f3288b7/utilities/transactions/lock/range/range_tree/range_tree_lock_manager.cc#L375